src/Microsoft.AspNetCore.Routing/RouteCollection.cs

```diff
@@ -138,9 +138,9 @@ private VirtualPathData NormalizeVirtualPath(VirtualPathData pathData)
 
             var url = pathData.VirtualPath;
 
-            if (!string.IsNullOrEmpty(url) && (_options.LowercaseUrls || _options.AppendTrailingSlash))
+            if (!string.IsNullOrEmpty(url) && (_options.LowercaseUrls || _options.LowercaseQueryStrings || _options.AppendTrailingSlash))
```

Comment:
`LowercaseQueryStrings` shouldn't appear here, that just applies to the query string.